packages/nimble-components/src/rich-text/models/tests/markdown-serializer.spec.ts

```diff
@@ -0,0 +1,188 @@
+
+describe('Markdown serializer', () => {
+    const mockEditor = new Editor({
+        element: document.createElement('div'),
+        extensions: [
+            Document,
+            Paragraph,
+            Text,
+            BulletList,
+            OrderedList,
+            ListItem,
+            Bold,
+            Italic
+        ]
+    });
+
+    function getNode(htmlString: string): Node {
+        mockEditor.commands.setContent(htmlString);
+        return mockEditor.state.doc;
+    }
+
+    afterAll(() => {
+        mockEditor.destroy();
+    });
+
+    describe('various supported nodes should be serialized to a markdown output', () => {
+        const supportedNodesMarks: {
+            name: string,
+            html: string,
+            markdown: string
+        }[] = [
+            {
+                name: 'Bold',
+                html: '<strong>Bold</strong>',
+                markdown: '**Bold**'
+            },
+            {
+                name: 'Italics',
+                html: '<em>Italics</em>',
+                markdown: '*Italics*'
+            },
+            {
+                name: 'Bold and Italics',
+                html: '<strong><em>Bold and Italics</em></strong>',
+                markdown: '***Bold and Italics***'
+            },
+            {
+                name: 'Numbered list',
+                html: '<ol><li><p>Numbered list</p></li></ol>',
+                markdown: '1. Numbered list'
+            },
+            {
+                name: 'Multiple numbered list',
+                html: '<ol><li><p>list 1</p></li><li><p>list 2</p></li></ol>',
+                markdown: '1. list 1\n\n2. list 2'
+            },
+            {
+                name: 'Numbered list with bold',
+                html: '<ol><li><p><strong>Numbered list with bold</strong></p></li></ol>',
+                markdown: '1. **Numbered list with bold**'
+            },
+            {
+                name: 'Numbered list with italics',
+                html: '<ol><li><p><em>Numbered list with italics</em></p></li></ol>',
+                markdown: '1. *Numbered list with italics*'
+            },
+            {
+                name: 'Bullet list',
+                html: '<ul><li><p>Bullet list</p></li></ul>',
+                markdown: '* Bullet list'
+            },
+            {
+                name: 'Multiple Bullet list',
+                html: '<ul><li><p>list 1</p></li><li><p>list 2</p></li></ul>',
+                markdown: '* list 1\n\n* list 2'
+            },
+            {
+                name: 'Bullet list with bold',
+                html: '<ul><li><p><strong>Bullet list with bold</strong></p></li></ul>',
+                markdown: '* **Bullet list with bold**'
+            },
+            {
+                name: 'Bullet list with italics',
+                html: '<ul><li><p><em>Bullet list with italics</em></p></li></ul>',
+                markdown: '* *Bullet list with italics*'
+            }
+        ];
```

Comment:
Outside of formatting test cases, consider adding some general markdown serialization test cases: 
1. [HTML Entities](https://developer.mozilla.org/en-US/docs/Glossary/Entity) should be converted to their corresponding markdown equivalent character as given in the table
2. Markdown syntax characters for formatting or lists like `*`, `_` etc. when coming in HTML content, should be escaped properly in the markdown serialized output.